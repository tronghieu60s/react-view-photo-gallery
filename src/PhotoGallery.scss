@mixin user-select-none {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@mixin square($size: 0) {
  width: $size;
  height: $size;
}

@mixin fixed($top: auto, $right: auto, $bottom: auto, $left: auto) {
  position: absolute;
  top: $top;
  right: $right;
  bottom: $bottom;
  left: $left;
}

@mixin absolute($top: auto, $right: auto, $bottom: auto, $left: auto) {
  position: absolute;
  top: $top;
  right: $right;
  bottom: $bottom;
  left: $left;
}

@mixin flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.gallery-container {
  @include user-select-none;
  @include fixed($top: 0, $left: 0);
  @include square(100%);
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  outline: none;
  z-index: 100000;
}

.gallery-bg {
  @include absolute(0, 0, 0, 0);
  opacity: 0.9;
  background: #1e1e1e;
}

.gallery-inner {
  @include absolute(0, 0, 0, 0);
  backdrop-filter: blur(5px);
  z-index: 110000;
}

.gallery-button {
  @include square(44px);
  line-height: 35px;
  font-size: 0.95em;
  background: rgba(30, 30, 30, 0.6);
  display: inline-block;
  padding: 10px;
  border: 0;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  svg {
    @include square(20px);
    font-size: inherit;
    path {
      fill: #ccc;
    }
  }

  &:hover svg path {
    fill: #fff;
  }
}

.gallery-info {
  @include absolute($top: 0, $left: 0);
  color: #ccc;
  font-size: 0.85em;
  height: 44px;
  line-height: 44px;
  padding: 0 10px;
}

.gallery-toolbar {
  @include absolute($top: 0, $right: 0);
  width: 308px;
  height: 44px;
}

.gallery-navigation {
  &__left,
  &__right {
    @include absolute($top: 50%, $left: 0);
    transform: translateY(-50%);
    margin: 0 1em;
    border-radius: 5px;
  }

  &__right {
    @include absolute($top: 50%, $right: 0);
  }
}

.gallery-stage {
  height: 100%;
  display: flex;
  flex-direction: row;
  overflow: hidden;

  &__slide {
    @include flex-center;
    @include square(100%);
  }

  &__content {
    @include square(300px);
    cursor: zoom-in;
  }

  &__image {
    @include square(100%);
  }
}

.gallery-caption {
  @include absolute();
  background: linear-gradient(
    to top,
    rgba(0, 0, 0, 0.85) 0%,
    rgba(0, 0, 0, 0.3) 50%,
    rgba(0, 0, 0, 0.15) 65%,
    rgba(0, 0, 0, 0.075) 75.5%,
    rgba(0, 0, 0, 0.037) 82.84999999999999%,
    rgba(0, 0, 0, 0.019) 88%,
    rgba(0, 0, 0, 0) 100%
  );
  bottom: 0;
  color: #eee;
  font-size: 14px;
  font-weight: 400;
  left: 0;
  line-height: 1.5;
  padding: 75px 44px 25px 44px;
  pointer-events: none;
  right: 0;
  text-align: center;
  z-index: 110000;

  &__body {
    max-height: 50vh;
    overflow: auto;
    pointer-events: all;
    h4,
    p {
      margin: 4px 0;
      word-wrap: break-word;
    }
  }
}
