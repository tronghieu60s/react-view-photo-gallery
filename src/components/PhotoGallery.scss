$thumb-size-mobile: 75px;
$thumb-size-tablet: 120px;
$thumb-size-desktop: 150px;

// -----------------------------
// Style Common Mixin ----------
// -----------------------------

@mixin visible {
  opacity: 1;
  visibility: visible;
}

@mixin hidden {
  opacity: 0;
  visibility: hidden;
}

@mixin square($size: 0) {
  width: $size;
  height: $size;
}

@mixin inset($top: auto, $right: auto, $bottom: auto, $left: auto) {
  top: $top;
  right: $right;
  bottom: $bottom;
  left: $left;
}

@mixin fixed($top: auto, $right: auto, $bottom: auto, $left: auto) {
  position: fixed;
  @include inset($top, $right, $bottom, $left);
}

@mixin absolute($top: auto, $right: auto, $bottom: auto, $left: auto) {
  position: absolute;
  @include inset($top, $right, $bottom, $left);
}

@mixin flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}

@mixin user-select-none {
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

// -----------------------------
// Style Global Gallery --------
// -----------------------------
.gl-d-none {
  display: none;
}

// -----------------------------
// Style Photo Gallery ---------
// -----------------------------
.gl-container {
  @include hidden();
  @include user-select-none;
  @include fixed($top: 0, $left: 0);
  @include square(100%);
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  outline: none;
  z-index: 100000;
  transition: 0.4s;
  &.active {
    @include visible();
  }
  &.thumbs-active {
    .gl-inner {
      bottom: $thumb-size-desktop;
      @media (max-width: 768px) {
        bottom: $thumb-size-tablet;
      }
      @media (max-width: 425px) {
        bottom: $thumb-size-mobile;
      }
    }
    .gl-thumbs {
      @include visible();
    }
  }
}

.gl-bg {
  @include absolute(0, 0, 0, 0);
  background: #1e1e1e;
  opacity: 0.9;
}

.gl-inner {
  @include absolute(0, 0, 0, 0);
  backdrop-filter: blur(5px);
  overflow: hidden;
  z-index: 1000;
}

.gl-button {
  @include square(44px);

  display: inline-block;

  font-size: 0.95em;
  line-height: 35px;
  background: rgba(30, 30, 30, 0.6);

  padding: 10px;
  border: 0;
  border-radius: 0;
  box-shadow: none;

  cursor: pointer;
  svg {
    @include square(20px);
    font-size: inherit;
    path {
      fill: #ccc;
    }
  }
  &:hover svg path {
    fill: #fff;
  }
}

.gl-info {
  @include absolute($top: 0, $left: 0);
  height: 44px;

  font-size: 0.85em;
  line-height: 44px;
  color: #ccc;

  padding: 0 10px;
}

.gl-toolbar {
  @include absolute($top: 0, $right: 0);
  width: 264px;
  height: 44px;
  z-index: 10;
}

.gl-navigation {
  &__left,
  &__right {
    @include absolute($top: 50%, $left: 0);
    border-radius: 5px;
    margin: 0 1em;
    z-index: 10;
    transform: translateY(-50%);
  }
  &__right {
    @include absolute($top: 50%, $right: 0);
  }
}

.gl-stage {
  @include square(100%);
  @include absolute($left: 0);
  overflow: hidden;
  transition: 1s cubic-bezier(0.075, 0.82, 0.165, 1);
  &__item {
    width: 100%;
    &__content {
      width: 300px;
      padding-bottom: 60px;
      margin: auto;
      cursor: auto;
    }
  }
  &__slide {
    @include flex-center;
    @include square(100%);
  }
  &__image {
    @include square(100%);
    border-radius: 5px;
  }
}

.gl-caption {
  @include absolute($left: 0, $right: 0, $bottom: 0);

  text-align: center;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  color: #eee;
  background: linear-gradient(
    to top,
    rgba(0, 0, 0, 0.85) 0%,
    rgba(0, 0, 0, 0.3) 50%,
    rgba(0, 0, 0, 0.15) 65%,
    rgba(0, 0, 0, 0.075) 75.5%,
    rgba(0, 0, 0, 0.037) 82.84999999999999%,
    rgba(0, 0, 0, 0.019) 88%,
    rgba(0, 0, 0, 0) 100%
  );

  padding: 75px 44px 25px 44px;
  pointer-events: none;
  z-index: 110000;
  &__body {
    max-height: 50vh;
    overflow: auto;
    pointer-events: all;
    h4,
    span {
      min-height: 20px;
      margin: 4px 0;
      word-wrap: break-word;
    }
  }
}

.gl-thumbs {
  @include hidden();
  @include absolute(auto, 0, 0, 0);
  @include square(auto);
  background: rgba(0, 0, 0, 0.3);

  padding: 0 10px;
  box-sizing: border-box;
  scroll-behavior: smooth;

  overflow-x: auto;
  z-index: 110000;
  transition: 0.4s;
  &__list {
    @include square(100%);
    position: relative;
    display: flex;
    list-style: none;

    font-size: 0;
    white-space: nowrap;

    padding: 0;
    margin: 0;
    overflow-x: hidden;
    overflow-y: auto;
    &__item {
      @include square($thumb-size-desktop);
      position: relative;
      backface-visibility: hidden;
      background-color: rgba(0, 0, 0, 0.1);
      background-position: center center;
      background-repeat: no-repeat;
      background-size: cover;

      border-radius: 5px;
      padding: 0;
      margin: 4px;

      float: left;
      outline: none;
      overflow: hidden;
      cursor: pointer;
      @media (max-width: 768px) {
        @include square($thumb-size-tablet);
      }
      @media (max-width: 425px) {
        @include square($thumb-size-mobile);
      }
      &::before {
        @include absolute(0, 0, 0, 0);
        content: '';
      }
      &.active {
        &::before {
          border: 2px solid #eee;
          background: radial-gradient(transparent, rgba(0, 0, 0, 0.75));
        }
      }
    }
  }
}

.gl-progress {
  @include absolute($top: 0, $right: 0, $left: 0);
  height: 2px;
  background: #f2930d;

  transform: scaleX(0);
  transform-origin: 0;
  transition-property: transform;
  transition-timing-function: linear;
  z-index: 110000;
}

// -----------------------------
//  Style Scroll Bar -----------
// -----------------------------
::-webkit-scrollbar {
  @include square(4px);
  &-track {
    background: #f4f5f7;
  }
  &-thumb {
    border-radius: 2px;
    background: #2b2e34;
    &:hover {
      background: #555;
    }
  }
}
