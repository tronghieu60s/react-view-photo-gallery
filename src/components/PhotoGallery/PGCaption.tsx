import { ImageType } from "../../common/types";

type Props = {
  currentImage: ImageType;
};

export default function PGCaption(props: Props) {
  const { currentImage } = props;
  const { src, name, caption } = currentImage;
  const nameGl = name || src.split("/").pop();
  const captionGl = caption || new Date().toUTCString();
  return (
    <div className="gl-caption">
      <div className="gl-caption__body">
        <h4>{nameGl}</h4>
        <span>{captionGl}</span>
      </div>
    </div>
  );
}
