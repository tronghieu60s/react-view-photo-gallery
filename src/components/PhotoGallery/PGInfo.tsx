type Props = {
  numberOfImages: number;
  currentImageIndex: number;
};

export default function PGInfo(props: Props) {
  const { numberOfImages, currentImageIndex } = props;
  return (
    <div className="gl-info">
      <span>{currentImageIndex + 1}</span> / <span>{numberOfImages}</span>
    </div>
  );
}
