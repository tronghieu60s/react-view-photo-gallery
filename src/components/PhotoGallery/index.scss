$thumb-size: 150px;
$thumb-size-mobile: 75px;

// -----------------------------
// Style Common Mixin ----------
// -----------------------------

@mixin user-select-none {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@mixin visible {
  visibility: visible;
  opacity: 1;
}

@mixin hidden {
  visibility: hidden;
  opacity: 0;
}

@mixin square($size: 0) {
  width: $size;
  height: $size;
}

@mixin inset($top: auto, $right: auto, $bottom: auto, $left: auto) {
  top: $top;
  right: $right;
  bottom: $bottom;
  left: $left;
}

@mixin fixed($top: auto, $right: auto, $bottom: auto, $left: auto) {
  position: fixed;
  @include inset($top, $right, $bottom, $left);
}

@mixin absolute($top: auto, $right: auto, $bottom: auto, $left: auto) {
  position: absolute;
  @include inset($top, $right, $bottom, $left);
}

@mixin flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}

// -----------------------------
// Style Global Gallery --------
// -----------------------------
.gl-d-none {
  display: none;
}

// -----------------------------
// Style Photo Gallery ---------
// -----------------------------
.gl-container {
  @include user-select-none;
  @include fixed($top: 0, $left: 0);
  @include square(100%);
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  outline: none;
  z-index: 100000;
  transition: 0.4s;
  &-hidden {
    @include hidden();
  }
}

.gl-bg {
  @include absolute(0, 0, 0, 0);
  opacity: 0.9;
  background: #1e1e1e;
}

.gl-inner {
  @include absolute(0, 0, 0, 0);
  backdrop-filter: blur(5px);
  overflow: hidden;
  z-index: 110000;
}

.gl-button {
  @include square(44px);
  line-height: 35px;
  font-size: 0.95em;
  background: rgba(30, 30, 30, 0.6);
  display: inline-block;
  padding: 10px;
  border: 0;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  svg {
    @include square(20px);
    font-size: inherit;
    path {
      fill: #ccc;
    }
  }
  &:hover svg path {
    fill: #fff;
  }
}

.gl-info {
  @include absolute($top: 0, $left: 0);
  color: #ccc;
  font-size: 0.85em;
  height: 44px;
  line-height: 44px;
  padding: 0 10px;
}

.gl-toolbar {
  @include absolute($top: 0, $right: 0);
  width: 264px;
  height: 44px;
}

.gl-navigation {
  &__left,
  &__right {
    @include absolute($top: 50%, $left: 0);
    transform: translateY(-50%);
    margin: 0 1em;
    border-radius: 5px;
  }
  &__right {
    @include absolute($top: 50%, $right: 0);
  }
}

.gl-stage {
  height: 100%;
  display: flex;
  flex-direction: row;
  overflow: hidden;
  &__slide {
    @include flex-center;
    width: 100%;
    height: 100%;
  }
  &__content {
    width: 300px;
    padding-bottom: 60px;
    cursor: auto;
  }
  &__image {
    @include square(100%);
    border-radius: 5px;
  }
}

.gl-caption {
  @include absolute($left: 0, $right: 0);
  background: linear-gradient(
    to top,
    rgba(0, 0, 0, 0.85) 0%,
    rgba(0, 0, 0, 0.3) 50%,
    rgba(0, 0, 0, 0.15) 65%,
    rgba(0, 0, 0, 0.075) 75.5%,
    rgba(0, 0, 0, 0.037) 82.84999999999999%,
    rgba(0, 0, 0, 0.019) 88%,
    rgba(0, 0, 0, 0) 100%
  );
  bottom: 0;
  color: #eee;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  padding: 75px 44px 25px 44px;
  pointer-events: none;
  text-align: center;
  z-index: 110000;
  &__body {
    max-height: 50vh;
    overflow: auto;
    pointer-events: all;
    h4,
    p {
      min-height: 20px;
      margin: 4px 0;
      word-wrap: break-word;
    }
  }
}

.gl-show-thumbs .gl-thumbs {
  @include visible();
}

.gl-thumbs {
  @include hidden();
  @include absolute(auto, 0, 0, 0);
  width: auto;
  height: auto;
  padding: 0 10px;
  box-sizing: border-box;
  background: rgba(0, 0, 0, 0.3);
  scroll-behavior: smooth;
  overflow-x: auto;
  z-index: 110000;
  transition: 0.4s;
  &__list {
    @include square(100%);
    font-size: 0;
    list-style: none;
    white-space: nowrap;
    position: relative;
    margin: 0;
    padding: 0;
    overflow-x: hidden;
    overflow-y: auto;
  }
  &__item {
    @include square($thumb-size);
    position: relative;
    backface-visibility: hidden;
    background-color: rgba(0, 0, 0, 0.1);
    background-position: center center;
    background-repeat: no-repeat;
    background-size: cover;
    cursor: pointer;
    float: left;
    outline: none;
    overflow: hidden;
    margin: 4px;
    padding: 0;
    border-radius: 5px;
    &::before {
      @include absolute(0, 0, 0, 0);
      content: "";
    }
    &.active {
      &::before {
        border: 2px solid #eee;
        background: radial-gradient(transparent, rgba(0, 0, 0, 0.75));
      }
    }
  }
}

.gl-progress {
  @include absolute($top: 0, $right: 0, $left: 0);
  background: #f2930d;
  height: 2px;
  transform: scaleX(0);
  transform-origin: 0;
  transition-property: transform;
  transition-timing-function: linear;
  z-index: 110000;
}

// -----------------------------
//  Style Scroll Bar -----------
// -----------------------------
::-webkit-scrollbar {
  width: 4px;
  height: 4px;
  &-track {
    background: #f4f5f7;
  }
  &-thumb {
    border-radius: 2px;
    background: #2b2e34;
    &:hover {
      background: #555;
    }
  }
}
