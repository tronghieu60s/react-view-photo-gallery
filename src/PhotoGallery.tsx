import "./PhotoGallery.scss";

export default function PhotoGallery() {
  return (
    <div className="gallery-container">
      <div className="gallery-bg"></div>
      <div className="gallery-inner">
        <div className="gallery-inner-top">
          <div className="gallery-info">
            <span>1</span>&nbsp;/&nbsp;<span>2</span>
          </div>
          <div className="gallery-toolbar">
            <button className="gallery-button">
              <i className="fa fa-search-plus" aria-hidden="true"></i>
            </button>
            <button className="gallery-button">
              <i className="fa fa-external-link" aria-hidden="true"></i>
            </button>
            <button className="gallery-button">
              <i className="fa fa-arrows-alt" aria-hidden="true"></i>
            </button>
            <button className="gallery-button">
              <i className="fa fa-play" aria-hidden="true"></i>
            </button>
            <button className="gallery-button">
              <i className="fa fa-download" aria-hidden="true"></i>
            </button>
            <button className="gallery-button">
              <i className="fa fa-th" aria-hidden="true"></i>
            </button>
            <button className="gallery-button">
              <i className="fa fa-times" aria-hidden="true"></i>
            </button>
          </div>
        </div>
        <div className="gallery-inner-navigation">
          <button className="gallery-button">
            <i className="fa fa-arrow-left" aria-hidden="true"></i>
          </button>
          <button className="gallery-button">
            <i className="fa fa-arrow-right" aria-hidden="true"></i>
          </button>
        </div>
        {/* <div className="gallery-stage">
          <div className="gallery-slide gallery-slide--image gallery-slide--current gallery-slide--complete">
            <div className="gallery-content">
              <img
                src="https://i.imgur.com/QQCkPzN.jpeg"
                alt=""
                className="gallery-image"
              />
            </div>
          </div>
        </div> */}
        {/* <div className="gallery-caption gallery-caption--separate">
          <div className="gallery-caption__body">
            <h4>19.jpg</h4>
            <p>PhotoGallery Â· 8/8/21 at 16:35</p>
          </div>
        </div> */}
      </div>
      <div className="gallery-thumbs gallery-thumbs-x">
        <div className="gallery-thumbs__list">
          <a
            href=""
            style={{
              backgroundImage: "url(https://i.imgur.com/QQCkPzN.jpeg)",
            }}
          ></a>
          <a
            href=""
            style={{
              backgroundImage: "url(https://i.imgur.com/QQCkPzN.jpeg)",
            }}
            className="gallery-thumbs-active"
          ></a>
        </div>
      </div>
    </div>
  );
}
