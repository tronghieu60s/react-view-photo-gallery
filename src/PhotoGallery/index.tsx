import GalleryCaption from "./GalleryCaption";
import GalleryInfo from "./GalleryInfo";
import GalleryNavigation from "./GalleryNavigation";
import GalleryStage from "./GalleryStage";
import GalleryThumbs from "./GalleryThumbs";
import GalleryToolbar from "./GalleryToolbar";
import "./PhotoGallery.scss";

export default function PhotoGallery() {
  return (
    <div className="gallery-container gallery-container-show-thumbs">
      <div className="gallery-bg"></div>
      <div className="gallery-inner">
        <GalleryInfo />
        <GalleryToolbar />
        <GalleryNavigation />
        <GalleryStage />
        <GalleryCaption />
      </div>
      <GalleryThumbs />
    </div>
  );
}
