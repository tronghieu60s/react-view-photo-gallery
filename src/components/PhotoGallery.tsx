import { useEffect, useState } from "react";
import {
  ParametersProps,
  ParametersProvider,
} from "../contexts/ParametersContext";
import { getFileExtension } from "../helpers/commonFunctions";
import "./PhotoGallery.scss";
import PGMain from "./PhotoGallery/PGMain";

const SLIDE_SHOW_MILLISECONDS = 3000;

export default function PhotoGallery(props: ParametersProps) {
  const { images } = props;
  const [isFullscreen, setIsFullscreen] = useState(false);
  const [isSlideshow, setIsSlideshow] = useState(false);
  const [isOpenThumbs, setIsOpenThumbs] = useState(false);
  const [currentImageIndex, setCurrentImageIndex] = useState(0);
  const [currentProgress, setCurrentProgress] = useState(0);

  useEffect(() => {
    if (isSlideshow) {
      setCurrentProgress(3000);
      const slideshow = setInterval(() => {
        setCurrentProgress(0);
        setTimeout(onNextImage, 100);
      }, SLIDE_SHOW_MILLISECONDS);
      return () => clearInterval(slideshow);
    }
    setCurrentProgress(0);
  }, [isSlideshow, currentImageIndex]);

  const onPreviousImage = () => {
    if (currentImageIndex > 0) {
      setCurrentImageIndex(currentImageIndex - 1);
    } else {
      setCurrentImageIndex(images.length - 1);
    }
  };

  const onNextImage = () => {
    if (currentImageIndex < images.length - 1) {
      setCurrentImageIndex(currentImageIndex + 1);
    } else {
      setCurrentImageIndex(0);
    }
  };
  
  const onDownloadImage = () => {
    const { src } = images[currentImageIndex];
    const fileExtension = getFileExtension(src || "");
    fetch(src)
      .then((res) => res.blob())
      .then((blob) => {
        const href = window.URL.createObjectURL(blob);
        const link = document.createElement("a");
        link.href = href;
        link.setAttribute("download", `${Date.now()}.${fileExtension}`);
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
      })
      .catch((_) => alert("Cannot Download File!"));
  };

  const onOpenExternalLink = () => {
    window.open(images[currentImageIndex].src, "_blank");
  }

  const onToggleFullScreen = () => {
    if (!document.fullscreenElement || !isFullscreen) {
      document.documentElement.requestFullscreen();
      setIsFullscreen(true);
    } else {
      if (document.exitFullscreen) {
        document.exitFullscreen();
        setIsFullscreen(false);
      }
    }
  };

  return (
    <ParametersProvider {...props}>
      <PGMain
        images={images}
        numberOfImages={images.length}
        currentImage={images[currentImageIndex]}
        currentImageIndex={currentImageIndex}
        currentProgress={currentProgress}
        /* Actions */
        onPreviousImage={onPreviousImage}
        onNextImage={onNextImage}
        onSetCurrentImageIndex={(index) => setCurrentImageIndex(index)}
        /* Action Button */
        onDownloadImage={onDownloadImage}
        onOpenExternalLink={onOpenExternalLink}
        isFullscreen={isFullscreen}
        onToggleFullScreen={onToggleFullScreen}
        isSlideshow={isSlideshow}
        onToggleSlideshow={() => setIsSlideshow(!isSlideshow)}
        isOpenThumbs={isOpenThumbs}
        onToggleThumbs={() => setIsOpenThumbs(!isOpenThumbs)}
      />
    </ParametersProvider>
  );
}
