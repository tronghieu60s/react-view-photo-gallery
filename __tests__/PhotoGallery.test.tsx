import React from 'react';
import { mount, shallow } from 'enzyme';

import PhotoGallery, { PhotoGalleryProps } from '../dist/index';

const defaultProps: PhotoGalleryProps = {
  show: true,
  onShow: () => {},
  images: [
    { src: 'https://via.placeholder.com/500x500' },
    { src: 'https://via.placeholder.com/1000x1000', name: 'Amazing Name' },
    {
      src: 'https://via.placeholder.com/1200x1800',
      name: 'Amazing Name',
      caption: 'Amazing Caption',
    },
  ],
};

describe('<PhotoGallery />', () => {
  test('Render the component <PhotoGallery />', () => {
    expect(shallow(<PhotoGallery {...defaultProps} />)).toMatchSnapshot();
  });
  test('Count items in the component <PhotoGallery />', () => {
    const component = mount(<PhotoGallery {...defaultProps} />);
    expect(component.find('.gl-bg').length).toBe(1);
    expect(component.find('.gl-caption').length).toBe(1);
    expect(component.find('.gl-info').length).toBe(1);
    expect(component.find('.gl-navigation').length).toBe(1);
    expect(component.find('.gl-progress').length).toBe(1);
    expect(component.find('.gl-progress').length).toBe(1);
    expect(component.find('.gl-stage').length).toBe(1);
    expect(component.find('.gl-thumbs').length).toBe(1);
    expect(component.find('.gl-thumbs').length).toBe(1);
    expect(component.find('.gl-toolbar').length).toBe(1);
  });
  test('Info item in the component <PhotoGallery />', () => {
    const component = mount(<PhotoGallery {...defaultProps} />);
    /**
     * test show info of items,
     * click next and previous button for change value
     */
    expect(component.find('.gl-info').length).toBe(1);
    expect(component.find('.gl-info').text()).toEqual(`1 / 3`);
    component.find('.gl-navigation__right').simulate('click');
    expect(component.find('.gl-info').text()).toEqual(`2 / 3`);
    component.find('.gl-navigation__right').simulate('click');
    component.find('.gl-navigation__right').simulate('click');
    expect(component.find('.gl-info').text()).toEqual(`1 / 3`);
    component.find('.gl-navigation__left').simulate('click');
    expect(component.find('.gl-info').text()).toEqual(`3 / 3`);
    component.find('.gl-navigation__left').simulate('click');
    expect(component.find('.gl-info').text()).toEqual(`2 / 3`);
  });
  test('Caption item in the component <PhotoGallery />', () => {
    /**
     * test show caption of items,
     * click next and previous button for change value
     */
    const component = mount(<PhotoGallery {...defaultProps} />);
    expect(component.find('.gl-caption').length).toBe(1);
    expect(component.find('.gl-caption__body h4').text()).toEqual(`500x500`);
    component.find('.gl-navigation__right').simulate('click');
    expect(component.find('.gl-caption__body h4').text()).toEqual(`Amazing Name`);
    component.find('.gl-navigation__right').simulate('click');
    expect(component.find('.gl-caption__body h4').text()).toEqual(`Amazing Name`);
    expect(component.find('.gl-caption__body span').text()).toEqual(`Amazing Caption`);
  });
});
