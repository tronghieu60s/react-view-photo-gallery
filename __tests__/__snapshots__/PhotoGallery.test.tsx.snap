// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoGallery /> Render the component <PhotoGallery /> 1`] = `ShallowWrapper {}`;
